### PR TITLE
DOC: typo fixes and rephrase suggestions

### PR DESCRIPTION
**Pull request by yoavram**
Great tutorial, I was looking exactly for these features! 
This is just a bunch of typo fixes and rephrase suggestions
I hope my suggestions are helpful.

### CODE REVIEW

File: doc/model.rst [hunk 1]
```diff
@@ -47,20 +47,20 @@ own.  We start with a simple definition the model function:
...
-arrays ``y`` and ``x``.  Using :func:`scipy.optimize.curve_fit` makes this
-easy to do, allowing us to do something like::
+We want to fit this objective function to data :math:`y(x)` represented by the
+arrays ``y`` and ``x``.  This can be done easily wit :func:`scipy.optimize.curve_fit`::
 
     >>> from scipy.optimize import curve_fit
     >>>
-    >>> x, y = read_data_from_somewhere(....)
+    >>> x = linspace(-10,10)
+    >>> y = y = gaussian(x, 2.33, 0.21, 1.51) + np.random.normal(0, 0.2, len(x))
```
Comment (tritemio) [inline doc/model.rst]: y = y

File: doc/intro.rst [hunk 2]
```diff
@@ -79,13 +79,13 @@ including:
 
 These shortcomings are really do solely to the use of traditional arrays of
 variables, as matches closely the implementation of the Fortran code.  The
-lmfit module overcomes these shortcomings by using a core reason for using
-Python -- objects.  The key concept for lmfit is to use :class:`Parameter`
+lmfit module overcomes these shortcomings by using objects -- a core reason for wokring with
+Python.  The key concept for lmfit is to use :class:`Parameter`
```
Comment (tritemio) [inline doc/intro.rst]: typo "wokring with"

**Comment by newville** [comment]
@yoavram   Thanks!  It's always really helpful to have another set of eyes for the documentation.

I'm inclined to merge this and then make some of the minor corrections such as @tritemio suggests.  Any objections to that?

**Comment by yoavram** [comment]
Of course not, sorry for introducing new typos when trying to fix old ones ;)
Thanks for lmfit, I'm going to use extensively, much more user friendly and I was looking for a way to bound and fix the params.

**Comment by newville** [comment]
@yoavram  -- Thanks!  And no problem introducing more typos, it's progress!

**Pull request merged**